src/BSON/Binary.c

```diff
@@ -367,6 +367,32 @@ static phongo_create_object_retval php_phongo_binary_create_object(zend_class_en
 #endif
 } /* }}} */
 
+static phongo_create_object_retval php_phongo_binary_clone_object(zval* object TSRMLS_DC) /* {{{ */
+{
+	php_phongo_binary_t* intern;
+	php_phongo_binary_t* new_intern;
+	phongo_create_object_retval new_object;
+
+	intern = Z_BINARY_OBJ_P(object);
+	new_object = php_phongo_binary_create_object(Z_OBJCE_P(object) TSRMLS_CC);
+
+#if PHP_VERSION_ID >= 70000
+	new_intern = Z_OBJ_BINARY(new_object);
+	zend_objects_clone_members(&new_intern->std, &intern->std TSRMLS_CC);
```

Comment:
Is this just relevant in the event that users have assigned their own public properties to the BSON object before attempting to clone it? I found a reference to this in [`clone_obj`](https://wiki.php.net/internals/engine/objects#clone_obj) documentation, but not much else.

Reply:
There is some sparse documentation to be found ([link](https://wiki.php.net/internals/engine/objects)):
```
/* Copies the properties of the old_object and calls the class entry
 * clone handler. Used in the implementation of zend_objects_clone_obj
 * In PHP > 5.3, it also initializes the properties before. */
```

This may be moot since we don't store properties on the object, but I thought it'd be good practice to call this regardless. This would indeed copy user-assigned properties when cloning BSON objects, for better or worse.